src/commands/zombienet/init.ts

```diff
@@ -0,0 +1,95 @@
+import path from "node:path";
+import { Flags } from "@oclif/core";
+import { SwankyCommand } from "../../lib/swankyCommand.js";
+import {
+  buildZombienetConfigFromBinaries,
+  copyZombienetTemplateFile,
+  downloadZombienetBinaries,
+  getTemplates,
+  osCheck,
+  Spinner,
+} from "../../lib/index.js";
+import { pathExistsSync } from "fs-extra/esm";
+import { zombienet, zombienetBinariesList } from "../../lib/zombienetInfo.js";
+
+export const zombienetConfig = "zombienet.config.toml";
+
+export class InitZombienet extends SwankyCommand<typeof InitZombienet> {
+  static description = "Initialize Zombienet";
+
+  static flags = {
+    binaries: Flags.string({
+      char: "b",
+      multiple: true,
+      required: false,
+      options: zombienetBinariesList,
+      default: [],
+      description: "Binaries to install",
+    }),
+  };
+
+  async run(): Promise<void> {
+    const { flags } = await this.parse(InitZombienet);
+
+    const platform = osCheck().platform;
+    if (platform === "darwin") {
+      this.warn(`Note for MacOs users: Polkadot binary is not currently supported for MacOs.
+As a result users of MacOS need to clone the Polkadot repo (https://github.com/paritytech/polkadot), create a release and add it in your PATH manually (setup will advice you so as well). Check the official zombienet documentation for manual settings: https://paritytech.github.io/zombienet/.`);
+    }
+
+    const projectPath = path.resolve();
+    if (pathExistsSync(path.resolve(projectPath, "zombienet", "bin", "zombienet"))) {
+      this.error("Zombienet config already initialized");
+    }
+
+    const spinner = new Spinner(flags.verbose);
+
+    this.swankyConfig.zombienet = {
+      version: zombienet.version,
+      downloadUrl: zombienet.downloadUrl,
+      binaries: {},
+    };
+
+    if (!flags.binaries.includes("polkadot")) {
+      flags.binaries.push("polkadot");
+    }
+
+    for (const binaryName of flags.binaries) {
+      if (platform === "darwin" && binaryName.startsWith("polkadot")) {
+        continue;
+      }
+      if (!Object.keys(zombienet.binaries).includes(binaryName)) {
+        this.error(`Binary ${binaryName} not found in Zombienet config`);
+      }
+      this.swankyConfig.zombienet.binaries[binaryName] = zombienet.binaries[binaryName as keyof typeof zombienet.binaries];
+    }
+
+    await this.storeConfig();
+
+    const zombienetTemplatePath = getTemplates().zombienetTemplatesPath;
+
+    const configPath = path.resolve(projectPath, "zombienet", "config");
+
+    if (flags.binaries.length < 2) {
```

Comment:
nit: stating this explicitly that `flags.binaries === ["polkadot"]` would clarify the code a tad

Reply:
nice suggestion, done ✅